test/bats/helpers/ln.bash

```diff
@@ -4,9 +4,38 @@ source $(dirname "$BASH_SOURCE")/_common.bash
 LND_FUNDING_TOKEN_NAME="lnd_funding"
 LND_FUNDING_PHONE="+16505554351"
 
+mempool_not_empty() {
+  local txid= [[ "$(bitcoin_cli getrawmempool | jq -r ".[0]")" != "null" ]]
+  [[ "$txid" != "null" ]] || exit 1
+}
+
+run_with_lnd() {
+  local func_name="$1"
+  shift  # This will shift away the function name, so $1 becomes the next argument
+
+  if [[ "$func_name" == "lnd_cli" ]]; then
+    lnd_cli "$@"
+  elif [[ "$func_name" == "lnd2_cli" ]]; then
+    lnd2_cli "$@"
+  else
+    echo "Invalid function name passed!" && exit 1
+  fi
+}
+
+close_partner_initiated_channels() {
+  ln_cli_name=$1
+
+  run_with_lnd "$ln_cli_name" listchannels \
+    | jq -r '.channels[] | select(.initiator != true) | .channel_point' \
+    | while read -r channel_point; do
+        funding_txid="${channel_point%%:*}"
+        run_with_lnd "$ln_cli_name" closechannel "$funding_txid"
+      done
+}
+
 lnds_init() {
   # Clean up any existing channels
-  lnd_cli closeallchannels || true
+  close_partner_initiated_channels lnd_cli || true
```

Comment:
where is this method used with lnd2_cli? or why the wrapper?

Reply:
Refactored it here to be more explicit. Ideally this is a function that closes all non-self-initiated channels from any of our internal nodes. It replaces the `closeAllChannels` usage because we can't yet account for channels closed that our internal node has initiated.

https://github.com/GaloyMoney/galoy/pull/3016/commits/fe16e45bf01bd072ae0a441fed31f8efcc679c89